librz/core/cbin.c

```diff
@@ -3591,60 +3669,67 @@ static int bin_classes(RzCore *r, PJ *pj, int mode) {
 		} else if (IS_MODE_CLASSDUMP(mode)) {
 			if (c) {
 				RzBinFile *bf = rz_bin_cur(r->bin);
-				if (bf && bf->o) {
-					if (IS_MODE_RZCMD(mode)) {
-						classdump_c(r, c);
-					} else if (bf->o->lang == RZ_BIN_NM_JAVA || (bf->o->info && bf->o->info->lang && strstr(bf->o->info->lang, "dalvik"))) {
-						classdump_java(r, c);
-					} else {
-						classdump_objc(r, c);
-					}
+				if (IS_MODE_RZCMD(mode)) {
+					classdump_c(r, c);
+				} else if (is_java_lang(bf)) {
+					classdump_java(r, c);
 				} else {
 					classdump_objc(r, c);
 				}
 			}
 		} else if (IS_MODE_RZCMD(mode)) {
-			char *n = __filterShell(name);
-			rz_cons_printf("\"f class.%s = 0x%" PFMT64x "\"\n", n, at_min);
-			free(n);
-			if (c->super) {
-				char *cn = c->name; // __filterShell (c->name);
-				char *su = c->super; // __filterShell (c->super);
-				rz_cons_printf("\"f super.%s.%s = %d\"\n",
-					cn, su, c->index);
-				// free (cn);
-				// free (su);
+			RzBinFile *bf = rz_bin_cur(r->bin);
+
+			// class
+			char *fn = rz_core_bin_class_build_flag_name(c);
+			if (fn) {
+				rz_cons_printf("\"f %s = 0x%" PFMT64x "\"\n", fn, at_min);
+				free(fn);
+			}
+
+			// super class
+			fn = rz_core_bin_super_build_flag_name(c);
+			if (fn) {
+				rz_cons_printf("\"f %s = %d\"\n", fn, c->index);
+				free(fn);
+			}
+
+			// class fields
+			rz_list_foreach (c->fields, iter2, f) {
+				char *fn = rz_core_bin_field_build_flag_name(c, f);
+				if (fn) {
+					rz_cons_printf("\"f %s = 0x%08" PFMT64x "\"\n", fn, f->vaddr);
+					free(fn);
+				}
 			}
+
+			// class methods
 			rz_list_foreach (c->methods, iter2, sym) {
 				char *fn = rz_core_bin_method_build_flag_name(c, sym);
 				if (fn) {
 					rz_cons_printf("\"f %s = 0x%" PFMT64x "\"\n", fn, sym->vaddr);
 					free(fn);
 				}
 			}
-			rz_list_foreach (c->fields, iter2, f) {
-				char *fn = rz_str_newf("field.%s.%s", c->name, f->name);
-				ut64 at = f->vaddr; //  sym->vaddr + (f->vaddr &  0xffff);
-				rz_cons_printf("\"f %s = 0x%08" PFMT64x "\"\n", fn, at);
-				free(fn);
-			}
 
 			// C struct
-			rz_cons_printf("td \"struct %s {", c->name);
-			if (rz_list_empty(c->fields)) {
-				// XXX workaround because we cant register empty structs yet
-				// XXX https://github.com/rizinorg/rizin/issues/16342
-				rz_cons_printf(" char empty[0];");
-			} else {
-				rz_list_foreach (c->fields, iter2, f) {
-					char *n = objc_name_toc(f->name);
-					char *t = objc_type_toc(f->type);
-					rz_cons_printf(" %s %s;", t, n);
-					free(t);
-					free(n);
+			if (!(bf->o->lang == RZ_BIN_NM_JAVA || (bf->o->info && bf->o->info->lang && strstr(bf->o->info->lang, "dalvik")))) {
+				rz_cons_printf("td \"struct %s {", c->name);
+				if (rz_list_empty(c->fields)) {
+					// XXX workaround because we cant register empty structs yet
```

Comment:
Are you sure? It was the old bug, now it works just fine:
```
[0x0000000]> td "struct Empty {}"
[0x0000000]> tsc Empty
struct Empty {};
[0x0000000]> 
```

Reply:
that is not my comment. i can remove that